synapse/rest/client/v2_alpha/sync.py

```diff
@@ -118,8 +120,14 @@ def on_GET(self, request):
             except:
                 filter = FilterCollection({})
 
+        if is_guest and filter.list_rooms() is None:
```

Comment:
Does it matter? I'm not sure if passing an empty list of rooms is invalid or not here.